citar.el

```diff
@@ -1078,7 +1078,7 @@ With prefix, rebuild the cache before offering candidates."
   (or (seq-some
        (lambda (opener)
          (funcall opener key entry)) citar-open-note-functions)
-      (funcall citar-create-note-function key entry)))
+      (funcall citar-file--open-note key entry)))
```

Comment:
I'm a little confused ATM; it's possible I'm just distracted, but here goes ...

The logic of the code is:

1. try to open using the `citar-open-note-functions`
2. if that doesn't work, then create a new note using the `citar-create-note-function`

But here, unless I'm misunderstanding something, you're instead just falling back to `citar--open-note`, even though that function is already included by default in `citar-open-note-functions`.

Can you clarify?

If you want that behavior, it might be good to change the suggestion on the config approach in the README, and use `add-to-list` instead?

Reply:
> EDIT: I see your note now on the issue about the 3 args. I need to think on that.

Seems the problem is that the default create-note function requires three args, but the `funcall` on it only includes two.

For whatever reason I can't figure out, I get no error!

Reply:
@bdarcus, Understood what you're trying to say!

However, as you mention, the issue is that if we use `orb-citar-edit-note` as a possible override for the `citar-open-note-functions`, then they must accept two arguments. I haven't looked into either ORB or Citar code deep enough to be able to suggest an alternative.

For the time being, I agree that suggesting `add-to-list` might be better than what I have done, while the Roam integration evolves in Citar in your following commits.

Reply:
What happens now if you specify the orb function for `citar-open-note-functions`, and then open a note for a reference that does not currently have a note?

My assumption/hope is that it calls the orb function, and creates the new note. If that's the case, that's a function call with two arguments, and it wouldn't get to the line I quote above?

Reply:
If I have `:immediate-finish t` on my template, then things work as expected, since the ORB capture process returns `t`. Otherwise, it will return `nil`, and Citar will try to call create note with 2 arguments instead of 3.

I now realise that for the time being, it may be enough to add the note about `:immediate-finish t` to the Readme file, while the integration with Roam improves also.

Reply:
That's good.

Do you want to drop the citar.el commit, then, I can merge the README change, and we can think a bit more about if the code needs a subtle refactor?

Reply:
Done :+1:

Reply:
One last little change:

I'm trying to keep the README one-sentence-per-line. Can you fix that please; add a line break between the two sentences?

Reply:
> I agree that suggesting `add-to-list` might be better than what I have done

Just a follow-up on this. In getting #609 working just now, I realize `add-to-list` wouldn't be right, as it won't stop at the first non-nil result. So if I'm opening a sub-file ref node, it will use org-roam to do that, and then after that use the default citar open function create a new file.